db/config.cc

```diff
@@ -591,6 +591,8 @@ db::config::config(std::shared_ptr<db::extensions> exts)
     , hinted_handoff_enabled(this, "hinted_handoff_enabled", value_status::Used, db::config::hinted_handoff_enabled_type(db::config::hinted_handoff_enabled_type::enabled_for_all_tag()),
         "Enable or disable hinted handoff. To enable per data center, add data center list. For example: hinted_handoff_enabled: DC1,DC2. A hint indicates that the write needs to be replayed to an unavailable node. "
         "Related information: About hinted handoff writes")
+    , max_hinted_handoff_concurrency_per_shard(this, "max_hinted_handoff_concurrency_per_shard", liveness::LiveUpdate, value_status::Used, 8,
+        "Maximum concurrency allowed for sending hints from a single shard. Setting this value to 0 does not disable hinted handoff, instead concurrency 1 will be used.")
```

Comment:
Better to make it per-node. Users don't expect per-shard configuration. The default can be 8*smp::count (or even the 4 that @vladzcloudius prefers).

Reply:
> or even the 4 that @vladzcloudius prefers
@avikivity, are you referring my `min_threshold` preference? ;)

If we make it per-node what shell we do if the user chooses a value less than the number of shards? Not multiple of shards? Fail to boot? 
If we choose to have a per-node value then the concurrency controller would rather be per-node too and this is going to be quite against a seastar mindset.

As unexpected as it is customers can deal with a per-shard parameter just alright.

Reply:
I'd vote for a per-shard parameter, too. I don't see how we can enforce a per-node limit without doing cross-shard operations. Besides, we already have a per-shard parameter in the configuration - `max_concurrent_requests_per_shard`.

Reply:
@piodul divide the concurrency by the number of shards (with std::max(1, ...))  and that's your concurrency limit.

We have a similar arrangement with commitlog size.

Reply:
@avikivity What about numbers not divisible by the shard count? Commitlog size is a number expressed in megabytes, so rounding down or up doesn't hurt. If somebody asks for a very low hint concurrency on the node (e.g. between `2*shard_count` or `3*shard_count`), rounding will have significant effect.

Reply:
It's okay if it's imperfect. I'm trying to avoid a situation where some config variables are per node, some per shard.

Reply:
OK. I changed it to a per-node parameter. Concurrency is now rounded up (as in the case of the commitlog size).